dm/syncer/dml_worker.go

```diff
@@ -117,24 +115,24 @@ func (w *DMLWorker) run() {
 
 	for j := range w.inCh {
 		metrics.QueueSizeGauge.WithLabelValues(w.task, "dml_worker_input", w.source).Set(float64(len(w.inCh)))
-		if j.tp == flush || j.tp == conflict {
-			if j.tp == conflict {
-				w.addCountFunc(false, adminQueueName, j.tp, 1, j.targetTable)
-			}
-			w.wg.Add(w.workerCount)
-			// flush for every DML queue
-			for i, jobCh := range jobChs {
-				startTime := time.Now()
-				jobCh <- j
-				metrics.AddJobDurationHistogram.WithLabelValues(j.tp.String(), w.task, queueBucketMapping[i], w.source).Observe(time.Since(startTime).Seconds())
-			}
-			w.wg.Wait()
-			if j.tp == conflict {
-				w.addCountFunc(true, adminQueueName, j.tp, 1, j.targetTable)
-			} else {
-				w.flushCh <- j
-			}
-		} else {
+		switch j.tp {
+		case flush:
+			w.addCountFunc(false, adminQueueName, j.tp, 1, j.targetTable)
+			w.sendJobToAllDmlQueue(j, jobChs, queueBucketMapping)
+			j.wg.Wait()
+			w.addCountFunc(true, adminQueueName, j.tp, 1, j.targetTable)
```

Comment:
for sync flush jobs, from this metrics we see that on one point in Grafana there's a job in adminQueue, and on another point the job is finished. The interval of points in Grafana is at lease 5s, maybe that's not accurate enough for observing the blocking time of flush?

maybe you're expected it a histogram like `metrics.AddJobDurationHistogram` in line 141?

Reply:
> record the total number of conlict/sync flush job have been committed?

`metrics.AddedJobsTotal` and `metrics.FinishedJobsTotal`?

> why we need to call w.addCountFunc(false and then w.addCountFunc(true here?

seems by `metrics.FinishedJobsTotal - metrics.AddedJobsTotal` we can get the jobs in the queue.

And my concern is that not about the number, but about the time cost of flush/conflict jobs. We may record the time and report to metrics like https://github.com/pingcap/ticdc/blob/f0a38a7aaf9f3b11a4d807da275b567642733f58/dm/syncer/syncer.go#L841-L842

We can do it in next PR.

Reply:
if i understand correctly, we would need to add extra metrics to recording the time spent on flush/conflict job here. And we don't need to refactor/change the current mechanism of recording total number of jobs.